d2bs/kolbot/libs/modules/Town.js

```diff
@@ -352,6 +352,26 @@
 		return true;
 	};
 
+	// Purchases and drinks potions in town.
+	Town.stackPotions = function(type, amount) {
+        const npc = Town.initNPC("Shop", "buyPotions");
+        for(var i = 0; i < amount; i++){
+			const npcPotion = Town.getPotion(npc, type);
+			if (npcPotion) {
+				npcPotion.buy(false);
+				const potion = me.getItem(npcPotion.classid);
+				if (potion) {
+					do {
+						if ((potion.mode === 0 && potion.location === 3) || potion.mode === 2) {
+							me.overhead('Stacking potion');
+							potion.interact();
+						}
+					} while (potion.getNext());
```

Comment:
Instead of the do { } while, it might be more clean if you use the me.getItems() function

e.g.
```javascript
me.getItems(npcPotion.classid)
	.filter(potion=>(potion.mode === 0 && potion.location === sdk.storage.Inventory) || potion.mode === 2)
	.forEach(function() {
		me.overhead('Stacking potion ('+i+'/'+amount+')');
		potion.interact();
	});

Reply:
Good idea. I ran into some trouble with `forEach` and breaking early as noted here: https://github.com/jaenster/d2bot-with-kolbot/pull/77/files#r354598108